feature-push-notifications/src/main/java/io/novafoundation/nova/feature_push_notifications/data/data/subscription/RealPushSubscriptionService.kt

```diff
@@ -101,58 +98,46 @@ class RealPushSubscriptionService(
         val govStateTracksDiff = CollectionDiffer.findDiff(
             oldItems = oldSettings.getGovernanceTracksFor { it.referendumUpdateEnabled },
-            newItems = governanceTracks,
+            newItems = newSettings?.getGovernanceTracksFor { it.referendumUpdateEnabled && pushEnabled } ?: emptyList(),
             forceUseNewItems = false
         )
         val newReferendaDiff = CollectionDiffer.findDiff(
             oldItems = oldSettings.getGovernanceTracksFor { it.newReferendaEnabled },
-            newItems = newSettings?.getGovernanceTracksFor { it.newReferendaEnabled } ?: emptyList(),
+            newItems = newSettings?.getGovernanceTracksFor { it.newReferendaEnabled && pushEnabled } ?: emptyList(),
             forceUseNewItems = false
         )
 
-        val deferreds = buildList<Deferred<Void>> {
-            val announcementsEnabled = newSettings?.announcementsEnabled ?: false
-            this += handleSubscription(announcementsEnabled && pushEnabled, "appUpdates")
+        val announcementsEnabled = newSettings?.announcementsEnabled ?: false
+        handleSubscription(announcementsEnabled && pushEnabled, "appUpdates")
 
-            this += govStateTracksDiff.added
-                .map { subscribeToTopic("${GOV_STATE_TOPIC_NAME}_${it.chainId}_${it.track}") }
-            this += govStateTracksDiff.removed
-                .map { unsubscribeFromTopic("${GOV_STATE_TOPIC_NAME}_${it.chainId}_${it.track}") }
-
-            this += newReferendaDiff.added
-                .map { subscribeToTopic("${NEW_REFERENDA_TOPIC_NAME}_${it.chainId}_${it.track}") }
-            this += newReferendaDiff.removed
-                .map { unsubscribeFromTopic("${NEW_REFERENDA_TOPIC_NAME}_${it.chainId}_${it.track}") }
-        }
+        govStateTracksDiff.added
+            .map { subscribeToTopic("${GOV_STATE_TOPIC_NAME}_${it.chainId}_${it.track}") }
+        govStateTracksDiff.removed
+            .map { unsubscribeFromTopic("${GOV_STATE_TOPIC_NAME}_${it.chainId}_${it.track}") }
 
-        deferreds.awaitAll()
+        newReferendaDiff.added
+            .map { subscribeToTopic("${NEW_REFERENDA_TOPIC_NAME}_${it.chainId}_${it.track}") }
+        newReferendaDiff.removed
+            .map { unsubscribeFromTopic("${NEW_REFERENDA_TOPIC_NAME}_${it.chainId}_${it.track}") }
     }
 
-    private fun handleSubscription(subscribe: Boolean, topic: String): Deferred<Void> {
+    private fun handleSubscription(subscribe: Boolean, topic: String) {
         return if (subscribe) {
             subscribeToTopic(topic)
         } else {
             unsubscribeFromTopic(topic)
         }
     }
 
-    private fun subscribeToTopic(topic: String): Deferred<Void> {
-        return Firebase.messaging.subscribeToTopic(topic)
-            .asDeferred()
+    private fun subscribeToTopic(topic: String) {
+        Firebase.messaging.subscribeToTopic(topic)
```

Comment:
Are we sure it is not lazy?

Reply:
Thank you for question! I checked it in Firebase SDK and we can see that `startTopicsSyncIfNecessary` is calling after scheduling the topic operation. So yes it's not lazy
<img width="486" alt="image" src="https://github.com/novasamatech/nova-wallet-android/assets/107959809/f54de4b1-82e9-489f-81da-51aa52dde736">